### PR TITLE
Add express-async-wrap

### PR DESCRIPTION
**Pull request by remyoudemans**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/express-async-wrap/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+declare function asyncWrap(
+  handler: (req: Request, res: Response, next: NextFunction) => Promise<void>
+): RequestHandler;
+
+declare function asyncWrap(
+  handler: (err: any, req: Request, res: Response, next: NextFunction) => Promise<void>,
+): ErrorRequestHandler;
+
+export = asyncWrap;
```
Comment (PranavSenthilnathan) [inline types/express-async-wrap/index.d.ts]: This [underlying package](https://github.com/Greenfields/express-async-wrap/blob/master/src/index.js#L6) is using a `default` export, so this declaration file should also do this. Your import in the test file will also have to change to the default import syntax.
```suggestion
export default asyncWrap;
```

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/express-async-wrap/index.d.ts (hunk 1 above)
Comment (remyoudemans) [reply types/express-async-wrap/index.d.ts]: Thanks for the feedback. Fixed!

**Comment by remyoudemans** [comment]
@PranavSenthilnathan - Are you happy with this now? Can it be merged?

**Review by PranavSenthilnathan — APPROVED** [review]

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**